concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/CredentialAttribute.java

```
@@ -0,0 +1,77 @@
+        TIMESTAMP;
+    }
+
+    private String value;
+    private CredentialAttributeType type;
+
+}
+
+class CredentialAttributieTypeDeserializer extends JsonDeserializer<CredentialAttribute> {
+
+    @Override
+    public CredentialAttribute deserialize(JsonParser p, DeserializationContext ctxt)
+            throws IOException, JsonProcessingException {
+        JsonNode reference = p.getCodec().readTree(p);
+        if (reference.isNumber()) {
+            return new CredentialAttribute(reference.asText(), CredentialAttributeType.INT);
+        } else if (reference.isTextual()) {
+            return new CredentialAttribute(reference.asText(), CredentialAttributeType.STRING);
+        } else if (reference.has("type") && reference.get("type").asText().equals("date-time")
+                && reference.has("timestamp")) {
+            return new CredentialAttribute(reference.get("timestamp").asText(), CredentialAttributeType.TIMESTAMP);
+        }
+        throw new JsonParseException(p, "Instantiation of Web3IdAtttribute failed due to invalid structure");
+    }
+}
+
+class CredentialAttributeTypeSerializer extends JsonSerializer<CredentialAttribute> {
+    @Override
+    public void serialize(CredentialAttribute value, JsonGenerator gen, SerializerProvider serializers)
+            throws IOException {
+                switch (value.getType()) {
+                    case INT:
+                        gen.writeNumber(value.getValue());
+                        break;
+                    case STRING:
+                        gen.writeString(value.getValue());
+                        break;
+                    case TIMESTAMP:
+                        gen.writeStartObject();
+                        gen.writeStringField("type", "date-time");
+                        gen.writeStringField("timestamp", value.getValue());
+                        gen.writeEndObject();
+                        break;
+                    default:
+                        break;
```

If we hit an unsupported case, then we should throw an error instead of doing nothing.